benchmarks/timeJava8/src/main/java/streams/Java8Streams.java

```diff
@@ -0,0 +1,204 @@
+        long ret = LongStream.of(v_P)
+                .filter(x -> (x & 0x15) != 0x11)
+                .filter(x -> (x & 0x10) == 0x10)
+                .filter(x -> (x & 0x30) != 0x10)
+                .filter(x -> (x & 0x10) == 0x10)
+                .filter(x -> (x & 0x15) != 0x10)
+                .filter(x -> (x & 0x10) == 0x10)
+                .filter(x -> (x & 0x30) != 0x10)
+                .filter(x -> (x & 0x15) == 0x10)
+                .reduce(0, (a, b) -> a + b);
+        bh.consume(ret);
+    }
+
+    @Benchmark
+    public void flatMap_take_primitive(Blackhole bh) {
+        long ret = LongStream.of(v_P)
+                .flatMap(x -> LongStream.of(vLo_P)
+                        .map(dP -> dP * x))
+                .limit(shortRangingFactor)
+                .reduce(0, (a, b) -> a + b);
+        bh.consume(ret);
+    }
+
+    @Benchmark
+    public void sum_boxed(Blackhole bh) {
+        Long ret = Stream.of(v_B)
+                .reduce(0L, (a, b) -> a + b);
+
+        bh.consume(ret);
+    }
+
+    @Benchmark
+    public void sumOfSquares_boxed(Blackhole bh) {
+        Long ret = Stream.of(v_B)
+                .map(d -> d * d)
+                .reduce(0L, (a, b) -> a + b);
+        bh.consume(ret);
+    }
+
+    @Benchmark
+    public void sumOfSquaresEven_boxed(Blackhole bh) {
+        Long ret = Stream.of(v_B)
+                .filter(x -> x % 2 == 0)
+                .map(x -> x * x)
+                .reduce(0L, (a, b) -> a + b);
+        bh.consume(ret);
+    }
+
+    @Benchmark
+    public void cart_boxed(Blackhole bh) {
+        Long ret = Stream.of(v_B)
+                .flatMap(d -> Stream.of(vLo_B)
+                        .map(dP -> dP * d))
+                .reduce(0L, (a, b) -> a + b);
+        bh.consume(ret);
+    }
+
+    @Benchmark
+    public void maps_boxed(Blackhole bh) {
+        Long ret = Stream.of(v_B)
+                .map(x -> x + (x & 0xD) + 0xCAFED00D)
+                .map(x -> x + (x & 0xE) + 0xD15EA5E)
+                .map(x -> x + (x & 0xA) + 0xDABBAD00)
+                .map(x -> x + (x & 0xD) + 0xDEADBAAD)
+                .map(x -> x + (x & 0xB) + 0xDEADDEAD)
+                .map(x -> x + (x & 0xE) + 0xDEADFA11)
+                .map(x -> x + (x & 0xE) + 0xFFBADD11)
+                .map(x -> x + (x & 0xF) + 0x4B1D)
+                .reduce(0L, (a, b) -> a + b);
+        bh.consume(ret);
+    }
+
+    @Benchmark
+    public void filters_boxed(Blackhole bh) {
+        Long ret = Stream.of(v_B)
+                .filter(x -> (x & 0x15) != 0x11)
+                .filter(x -> (x & 0x10) == 0x10)
+                .filter(x -> (x & 0x30) != 0x10)
+                .filter(x -> (x & 0x10) == 0x10)
+                .filter(x -> (x & 0x15) != 0x10)
+                .filter(x -> (x & 0x10) == 0x10)
+                .filter(x -> (x & 0x30) != 0x10)
+                .filter(x -> (x & 0x15) == 0x10)
+                .reduce(0L, (a, b) -> a + b);
+        bh.consume(ret);
```

Comment:
The idea is that you want to make optimizations fail. The code above would be good as long as the VM doesn't speculate on values(graal sometimes does).